### PR TITLE
validate lasttriggeredimageid hasn't changed before running build

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@ncdc @csrwng @soltysh ptal.

**Comment by bparees** [comment]
[test]

**Comment by ncdc** [comment]
LGTM until we can do the big refactor to model this like deployments as described in #2604

**Comment by bparees** [comment]
@ncdc do you want to use this to close 2604, or just update 2604 to a lower priority with some comments about what we did for now?

**Comment by ncdc** [comment]
@bparees I'd prefer to leave it open and lower the priority to 1 or 2.

File: pkg/build/generator/generator.go [hunk 1]
```diff
@@ -124,29 +124,37 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
...
 	if err := g.Client.UpdateBuildConfig(ctx, bc); err != nil {
+		glog.V(4).Infof("Failed to update BuildConfig %s/%s so no Build will be created", bc.Namespace, bc.Name)
+		return nil, err
+	}
+
+	// Ideally we would create the build *before* updating the BC to ensure that we don't set the LastTriggeredImageID
+	// on the BC and then fail to create the corresponding build, however doing things in that order allows for a race
+	// condition in which two builds get kicked off.  Doing it in this order ensures that we catch the race while
+	// updating the BC.
+	updatedBuild, err := g.createBuild(ctx, newBuild)
```
Comment (soltysh) [inline pkg/build/generator/generator.go]: You can just `return g.createBuild(ctx, newBuild)`

**Comment by soltysh** [comment]
Small nit, otherwise LGTM.

File: pkg/build/generator/generator.go (hunk 1 above)
Comment (bparees) [inline pkg/build/generator/generator.go]: good catch, thanks.

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Pull request merged**